### PR TITLE
Backend/refactor/move dashboard apis to dsl2

### PR DESCRIPTION
**Pull request by roman-bodavskiy**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [x] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added integration tests for my changes where possible
- [ ] I tested the changes using integration tests
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/app/dashboard/CommonAPIs/src/Dashboard/Common/Merchant.hs [hunk 1]
```diff
@@ -78,6 +74,28 @@ data MerchantEndpoint
   | DeleteSpecialLocationGateEndpoint
   | PostMerchantUpdateEndpoint
   | PostMerchantServiceConfigMapsUpdateEndpoint
+  | PostMerchantConfigCommonUpdateEndpoint
```
Comment (roman-bodavskiy) [inline Backend/app/dashboard/CommonAPIs/src/Dashboard/Common/Merchant.hs:77]: thinking to make this Endpoint enumeration auto generated also

File: Backend/app/dashboard/rider-dashboard/src/Domain/Action/RiderPlatform/Management/Merchant.hs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+postMerchantSpecialLocationUpsert merchantShortId opCity apiTokenInfo specialLocationId req = do
+  checkedMerchantId <- merchantCityAccessCheck merchantShortId apiTokenInfo.merchant.shortId opCity apiTokenInfo.city
+  transaction <- SharedLogic.Transaction.buildTransaction (Domain.Types.Transaction.MerchantAPI Dashboard.Common.Merchant.PostMerchantSpecialLocationUpsertEndpoint) (Kernel.Prelude.Just APP_BACKEND_MANAGEMENT) (Kernel.Prelude.Just apiTokenInfo) Kernel.Prelude.Nothing Kernel.Prelude.Nothing (Kernel.Prelude.Just req)
+  SharedLogic.Transaction.withTransactionStoring transaction $ (do error "Client call .merchantDSL.postMerchantSpecialLocationUpsert for separate helperApi couldn't be autogenerated. Logic yet to be decided" checkedMerchantId specialLocationId req)
+
+deleteMerchantSpecialLocationDelete :: (Kernel.Types.Id.ShortId Domain.Types.Merchant.Merchant -> Kernel.Types.Beckn.Context.City -> ApiTokenInfo -> Kernel.Types.Id.Id Lib.Types.SpecialLocation.SpecialLocation -> Environment.Flow Kernel.Types.APISuccess.APISuccess)
```
Comment (roman-bodavskiy) [inline Backend/app/dashboard/rider-dashboard/src/Domain/Action/RiderPlatform/Management/Merchant.hs:98]: Don't like how types generated with long qualified names. Thinking to add short synonyms for modules at least in domain handlers, and make it configurable in dhall

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src-read-only/API/Action/Dashboard/Fleet/Driver.hs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+postDriverFleetAddVehicle :: (Kernel.Types.Id.ShortId Domain.Types.Merchant.Merchant -> Kernel.Types.Beckn.Context.City -> Kernel.Prelude.Text -> Kernel.Prelude.Text -> Kernel.Prelude.Maybe Kernel.Prelude.Text -> API.Types.ProviderPlatform.Fleet.Driver.AddVehicleReq -> Environment.FlowHandler Kernel.Types.APISuccess.APISuccess)
+postDriverFleetAddVehicle a6 a5 a4 a3 a2 a1 = withFlowHandlerAPI $ Domain.Action.Dashboard.Fleet.Driver.postDriverFleetAddVehicle a6 a5 a4 a3 a2 a1
```
Comment (roman-bodavskiy) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src-read-only/API/Action/Dashboard/Fleet/Driver.hs:29]: will be changed to human readable params

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src-read-only/API/Action/Dashboard/Management/Booking.hs [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+module API.Action.Dashboard.Management.Booking
+  ( API.Types.ProviderPlatform.Management.Booking.API,
+    handler,
+  )
+where
+
+import qualified API.Types.ProviderPlatform.Management.Booking
+import qualified Dashboard.Common.Booking
+import qualified Domain.Action.Dashboard.Management.Booking as Domain.Action.Dashboard.Management.Booking
```
Comment (roman-bodavskiy) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src-read-only/API/Action/Dashboard/Management/Booking.hs:12]: TODO remove redundant as

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src-read-only/API/Action/Dashboard/Management/Driver.hs [hunk 5]
```diff
@@ -0,0 +1,110 @@
+{-# OPTIONS_GHC -Wno-orphans #-}
```
Comment (roman-bodavskiy) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src-read-only/API/Action/Dashboard/Management/Driver.hs:1]: TODO not required for this module

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src-read-only/API/Types/UI/VehicleDetails.hs [hunk 6]
```diff
@@ -9,10 +9,18 @@ import EulerHS.Prelude hiding (id)
...
+  deriving stock (Generic)
+  deriving anyclass (ToJSON, FromJSON, ToSchema)
 
-data VehicleModelsReq = VehicleModelsReq {make :: Data.Text.Text} deriving (Generic, ToJSON, FromJSON, ToSchema)
+data VehicleModelsReq = VehicleModelsReq {make :: Data.Text.Text}
+  deriving stock (Generic)
+  deriving anyclass (ToJSON, FromJSON, ToSchema)
 
-data VehicleModelsResp = VehicleModelsResp {models :: [Data.Text.Text]} deriving (Generic, ToJSON, FromJSON, ToSchema)
+data VehicleModelsResp = VehicleModelsResp {models :: [Data.Text.Text]}
```
Comment (roman-bodavskiy) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src-read-only/API/Types/UI/VehicleDetails.hs:24]: TODO newtype can be generated automatically if we have single field

File: Backend/app/dashboard/CommonAPIs/spec/ProviderPlatform/Management/API/Driver.yaml [hunk 7]
```diff
@@ -0,0 +1,385 @@
...
+  Version: Kernel.Types.Version
+  LatLong: Kernel.External.Maps.Types
+  City: Kernel.Types.Beckn.Context
+
+module: Driver
+
+apis:
+  - GET: # DriverDocumentsInfoAPI
+      endpoint: /documents/info
+      auth: ApiAuth DRIVER_OFFER_BPP_MANAGEMENT DRIVERS DOCUMENTS_INFO
```
Comment (roman-bodavskiy) [inline Backend/app/dashboard/CommonAPIs/spec/ProviderPlatform/Management/API/Driver.yaml:27]: TODO api name `DOCUMENTS_INFO`  and migration for dev api access also can be autogenerated

**Review by hkmangla — APPROVED** [review]

**Pull request merged**